js/router.js

```diff
@@ -485,11 +490,21 @@ export default class ObRouter extends Router {
       // not found page, otherwise display error.
       if (profileFetch.state() === 'rejected') {
         this.userNotFound(guid);
+        userPageFetchError = 'User Not Found';
       } else if (listingFetch.state() === 'rejected') {
         this.listingNotFound(deepRouteParts[0], `${guid}/${pageState}`);
+        userPageFetchError = 'Listing Not Found';
       }
     })
-      .always(() => (this.off(null, onWillRoute)));
+      .always(() => {
+        this.off(null, onWillRoute);
+        endEvent('UserPageLoad', {
+          ownPage: guid === app.profile.id,
+          tab: pageState,
+          listing: !!listingFetch,
+          errors: userPageFetchError || 'none',
```

Comment:
Countly doesn't record segments with blank values, so any event with an error will record "none" if it's successful. Otherwise you can't show a comparison of events with errors vs. events without errors.

Reply:
I'm not going to hold up merge for this, but I think it would have been more efficient to centralize this error: 'none' handling in the metric util module. That way you only write it once instead of every caller to endEvent, recordEvent, etc... having to follow this convention.

For example (metrics.js):

```
function parseSegmentation(segmentation = {}) {
  return {
    error: segmentation.error || 'none';
  }
}

export function endEvent(key, segmentation = {}) {
  const seg = parseSegmentation(segmentation);
  ...
}
```

Then instead of having to repeat something like this:

```
endEvent('Purchase', {
  errors: 'none',
 });
```

you could simplify to:

```
endEvent('Purchase');
```

Reply:
That's a good idea, though we might not always want an error segment on events. If that happened we could have a endEventNoError function to separate out the different use case.

Reply:
> we could have a endEventNoError function to separate out the different use case

I dunno if that's the best approach because then you would need a recordEventNoError function and maybe more.

I'm not sure I get the 'no error' use case when I thought the whole point of this was to make sure that the error is standardized. What would be a scenario where `error: none` would not work and it would be critical to not have the error key?

Anyways... we don't have to worry about it now.